src/components/Table/TableBase.tsx

```diff
@@ -0,0 +1,667 @@
+import { faSort } from '@fortawesome/free-solid-svg-icons/faSort';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import every from 'lodash/every';
+import filter from 'lodash/filter';
+import find from 'lodash/find';
+import includes from 'lodash/includes';
+import isEqual from 'lodash/isEqual';
+import isPlainObject from 'lodash/isPlainObject';
+import map from 'lodash/map';
+import reject from 'lodash/reject';
+import reverse from 'lodash/reverse';
+import some from 'lodash/some';
+import sortBy from 'lodash/sortBy';
+import * as React from 'react';
+import styled from 'styled-components';
+
+import { Button } from '../Button';
+
+// TODO: Remove explicit import and depend on provider instead.
+import theme from '../../theme';
+import { px } from '../../utils';
+import { Checkbox, CheckboxProps } from '../Checkbox';
+import { Pager } from '../Pager';
+import { CheckboxWrapper, TableColumn, TableRow } from './TableRow';
+
+const highlightStyle = `
+	background-color: ${theme.colors.info.light};
+`;
+
+const BaseTableWrapper = styled.div`
+	overflow-x: auto;
+	max-width: 100%;
+`;
+
+interface InternalTableBaseProps {
+	hasCheckbox: boolean;
+	hasRowClick: boolean;
+	hasGetRowRef: boolean;
+}
+
+const Base = styled.div<InternalTableBaseProps>`
+	display: table;
+	width: 100%;
+	border-spacing: 0;
+	border-bottom: 1px solid ${(props) => props.theme.colors.quartenary.main};
+
+	> [data-display='table-head'] {
+		display: table-header-group;
+		background-color: ${(props) => props.theme.colors.quartenary.semilight};
+
+		> [data-display='table-row'] {
+			display: table-row;
+
+			> [data-display='table-cell'] {
+				display: table-cell;
+				border-bottom: 1px solid
+					${(props) => props.theme.colors.quartenary.main};
+				text-align: left;
+				vertical-align: middle;
+				padding: 5px 20px;
+				font-size: ${(props) => px(props.theme.fontSizes[2])};
+			}
+
+			> [data-display='table-cell']:first-child {
+				padding-left: ${(props) => (props.hasCheckbox ? '20px' : '40px')};
+				${(props) => (props.hasCheckbox ? 'width: 60px' : '')};
+			}
```

Comment:
If we want to keep the first/last 40px padding when there is no checkbox (should we?) then we could refactor it a bit to make styled-components have to evaluate less functions:
```suggestion

			${(props) => props.hasCheckbox ?
				`> [data-display='table-cell']:first-child {
					width: 60px;
				}` :
				`> [data-display='table-cell']:first-child {
					padding-left: 40px;
				}

				> [data-display='table-cell']:last-child {
					padding-right: 40px;
				}`
			};
```

Reply:
:+1: